internal/provider/resource_bgp_session.go

```diff
@@ -381,13 +380,18 @@ func resourceBgpSessionRead(ctx context.Context, d *schema.ResourceData, m inter
 	if _, ok := d.GetOk("bfd_multiplier"); ok {
 		_ = d.Set("bfd_multiplier", bgp.BfdMultiplier)
 	}
-	if _, ok := d.GetOk("nat"); ok {
+	if bgp.Nat != nil {
 		nat := flattenNatConfiguration(bgp.Nat)
 		if err := d.Set("nat", nat); err != nil {
 			return diag.Errorf("error setting 'nat': %s", err)
 		}
 	}
-
+	if bgp.Prefixes != nil {
```

Comment:
I don't think we need to check `if bgp.Prefixes != nil` as those are required, there must be at least 2 prefixes (1 in, 1 out)